docs/design/proposals/CG-Common-Design-Goals.md

```diff
@@ -0,0 +1,92 @@
+# Design notes for grouping volumes for consistency group based DR
+
+## Grouping decision
+
+DRPC will carry an additional `spec.group` (or, `spec.async.group`) boolean,
+denoting if grouping of volumes is desired for consistency group (CG) based
+protection. VRG would also carry a similar spec field, that would be populated
+by the DRPC reconciler.
+
+- This is to facilitate upgrades from existing single PVC protection to
+  protecting PVCs via consistency groups.
+    - This field would be immutable, and hence existing DRPCs would default to
+      non-group behavior
+    - To shift to grouping behavior, the workload needs to be disabled for DR
+      protection and reenabled for DR protection with the grouping set to true
+- This also mandates that all PVCs SHOULD be grouped and hence enables VRG to
+  strictly group PVCs
+    - Any PVC that cannot be added to a CG would hence fail protection for the
+      entire workload
+
+## PVC grouping labels
+
+Both VolumeGroupSnapshot and VolumeGroupReplication definitions require a PVC
+label selector that can uniquely identify a group of PVCs that belong to the
+request. PVCs in this group should be an exact set belonging to a storage
+instance, such that the backing CSI implementation can snapshot/replicate the
+list of volume handles that are passed to it.
+
+This requires that PVCs that are to be protected, based on the VRG/DRPC
+`spec.pvcSelector` needs an additional label to separate this set into disjoint
+sets of PVCs, when they are provisioned by different storage instances.
+
+For the purposes of this grouping the `storageID`, that is matched across the
+StorageClass that the PVC is referring to with the VolumeGroupSnapshotClass,
+would be used to label the PVCs. (see storageID related design for properties of
+this label).
+
+The proposed label name is `ramendr.openshift.io/consistency-group`, with the
```

Comment:
@ELENAGER, your[ PR uses](https://github.com/ELENAGER/ramen/blob/e4e54e3aad59c0e612784447157a00f680ce1ee9/internal/controller/volumereplicationgroup_controller.go#L505) a different label name than the one proposed in this PR. 

Either way is fine with me; I am just ensuring that this PR or yours is updated to reflect the agreed naming.

Reply:
@BenamarMk I updated my PR to use the proposed label name.